### PR TITLE
Update OSCAR banner

### PR DESCRIPTION
**Pull request by paemurru**
In today's OSCAR meeting it was decided to decide by Monday which banner to use. Use the corresponding reaction to vote.

😄 Original OSCAR banner
vote_1_current

&nbsp;
&nbsp;
&nbsp;

🎉 Very slight change
vote_2_curved

&nbsp;
&nbsp;
&nbsp;

🚀 Julia-style, letters have vertical bars like in the current [OSCAR logo](https://avatars.githubusercontent.com/u/25104835?s=200&v=4) (according to the meeting, the current OSCAR logo might change soon)
vote_3_julia_mimic_oscar_logo

&nbsp;
&nbsp;
&nbsp;

❤️ Julia-style, clean version with no vertical bars
vote_4_julia_clean

### CODE REVIEW

**Comment by wdecker** [comment]
Von meinem iPhone gesendetAm 21.02.2024 um 18:37 schrieb Erik Paemurru ***@***.***>:﻿New banner
banner_new.png (view on web)
Old banner
banner_old.png (view on web)



You can view, comment on, or merge this pull request online at:
  https://github.com/oscar-system/Oscar.jl/pull/3410

Commit Summary

  b1a53be  New banner
  fc8e2d4  Raw strings to avoid '\' being treated as escaping


File Changes  (1 file)

  
    M
    src/Oscar.jl
    (14)
  


Patch Links:

  https://github.com/oscar-system/Oscar.jl/pull/3410.patch
  https://github.com/oscar-system/Oscar.jl/pull/3410.diff

**Comment by wdecker** [comment]
Much better, thanks!Von meinem iPhone gesendetAm 21.02.2024 um 19:03 schrieb Wolfram Decker ***@***.***>:﻿Von meinem iPhone gesendetAm 21.02.2024 um 18:37 schrieb Erik Paemurru ***@***.***>:﻿New banner
banner_new.png (view on web)
Old banner
banner_old.png (view on web)



You can view, comment on, or merge this pull request online at:
  https://github.com/oscar-system/Oscar.jl/pull/3410

Commit Summary

  b1a53be  New banner
  fc8e2d4  Raw strings to avoid '\' being treated as escaping


File Changes  (1 file)

  
    M
    src/Oscar.jl
    (14)
  


Patch Links:

  https://github.com/oscar-system/Oscar.jl/pull/3410.patch
  https://github.com/oscar-system/Oscar.jl/pull/3410.diff

**Review by fingolfin — DISMISSED** [review]
fine by me

**Comment by lgoettgens** [comment]
Please adapt the example output in the README.md file as well.

**Comment by paemurru** [comment]
Since we always call OSCAR from Julia, I thought perhaps it is nice to match the OSCAR banner style with Julia's, what do you think?
banner_figlet

**Comment by joschmitt** [comment]
I would be in favour of option # 3, if this is a vote. (Or anything else that at least partially resembles the "actual" logo from the website.)

**Comment by thofma** [comment]
I must have missed the discussion, where we decided that we want to update the banner.

Since other stakeholders might have an opinion, we can quickly discuss it tomorrow.

**Comment by paemurru** [comment]
Added more similarity with the OSCAR logo from the website.
banner_4_actual_logo

**Comment by aaruni96** [comment]
While we are taking the Julia banner as inspiration, it might be a nice idea to also add a link to the Oscar documentation along the Oscar banner.

**Comment by fingolfin** [comment]
I don't care much, but let me point out that one issue with the Julia banner is that it doesn't work well if the terminal width is below a certain threshold (there is even a PR for this: <https://github.com/JuliaLang/julia/pull/51811>). While the same holds for our current banner but the threshold is much smaller. 

Don't take this as an objection, just as something to be considered.

File: src/Oscar.jl [hunk 1]
```diff
@@ -39,21 +39,11 @@ if Sys.iswindows()
...
-  println(" -----    -----    -----   -     -  -     -  ")
-  println()
-  println("...combining (and extending) ANTIC, GAP, Polymake and Singular")
-  print("Version")
-  printstyled(" $VERSION_NUMBER ", color = :green)
-  print("... \n ... which comes with absolutely no warranty whatsoever")
-  println()
-  println("Type: '?Oscar' for more information")
-  println("(c) 2019-2024 by The OSCAR Development Team")
+  println(raw"  ___  ____   ____    _    _.__       ")
```
Comment (fingolfin) [inline src/Oscar.jl]: Shouldn't the first line also have the `|` if you want to match Julia? And no trailing whitespace?
```suggestion
  println(raw"  ___  ____   ____    _    _.__    |")
```

File: src/Oscar.jl [hunk 2]
```diff
@@ -39,21 +39,11 @@ if Sys.iswindows()
...
+  println(raw" / | \/ ___| / ___|  / \  | |_ \   |  Combining ANTIC, GAP, Polymake, Singular")
+  println(raw"| | | \_|_ \| |     / . \ | |_) |  |  Type \"?Oscar\" for more information")
+  println(raw"| |_| |___) | |___ / /_\ \| |_ <   |  ")
+  println(raw" \_|_/\____/ \____/_/ _ \_\_| \_\  |  " * "Version $VERSION_NUMBER")
```
Comment (fingolfin) [inline src/Oscar.jl]: You could also use a triple quoted string
```suggestion
  println(
    raw"""  ___  ____   ____    _    _.__    |  Combining ANTIC, GAP, Polymake, Singular
           / | \/ ___| / ___|  / \  | |_ \   |  Type "?Oscar" for more information
          | | | \_|_ \| |     / . \ | |_) |  |  Manual: https://docs.oscar-system.org
          | |_| |___) | |___ / /_\ \| |_ <   |
           \_|_/\____/ \____/_/ _ \_\_| \_\  |  Version """ * $VERSION_NUMBER)
```

File: src/Oscar.jl [hunk 3]
```diff
@@ -39,21 +39,12 @@ if Sys.iswindows()
...
-  print("... \n ... which comes with absolutely no warranty whatsoever")
-  println()
-  println("Type: '?Oscar' for more information")
-  println("(c) 2019-2024 by The OSCAR Development Team")
+  println(
+    raw"""  ___  ____   ____    _    _.__    |  Combining ANTIC, GAP, Polymake, Singular
+           / | \/ ___| / ___|  / \  | |_ \   |  Type "?Oscar" for more information
+          | | | \_|_ \| |     / . \ | |_) |  |  Manual: https://docs.oscar-system.org
+          | |_| |___) | |___ / /_\ \| |_ <   |
+           \_|_/\____/ \____/_/ _ \_\_| \_\  |  Version """ * $VERSION_NUMBER)
```
Comment (aaruni96) [inline src/Oscar.jl]: You have to enclose version number in string

```suggestion
           \_|_/\____/ \____/_/ _ \_\_| \_\  |  Version """ * "$VERSION_NUMBER")
```

File: src/Oscar.jl (hunk 1 above)
Comment (paemurru) [reply src/Oscar.jl]: Actually, the first line in the Julia banner just consists of the underscore character! To match it, I have made that the first line in the OSCAR banner consists of only underscore characters

**Comment by paemurru** [comment]
Looks like this now
banner_5_link

File: README.md [hunk 4]
```diff
@@ -31,21 +31,13 @@ in the OSCAR manual to learn more on how to contribute to OSCAR.
...
-|     |        |  |        |-----|  |   |
-|     |  |     |  |     |  |     |  |    |
- -----    -----    -----   -     -  -     -
-
-...combining (and extending) ANTIC, GAP, Polymake and Singular
-Version 1.1.0-DEV ...
-... which comes with absolutely no warranty whatsoever
-Type: '?Oscar' for more information
-(c) 2019-2024 by The OSCAR Development Team
+  ___  ____   ____    _    _.__
```
Comment (lgoettgens) [inline README.md]:
```suggestion
  ___  ____   ____    _    _ __
```
the single dot in the topmost row looks somehow off, it's too high

**Comment by joschmitt** [comment]
If this an ASCII art design competition I want to chuck in:
```
  _____  _____   _____     _     _  ___
 / _|_ \/  ___\ / ____|   / \   | | _  \
| /   \ | (___ | /       / . \  | | _) |
| |   | \__|  \| |      / / \ \ | | _ <
| \___/ |___)  | \____ / / = \ \| |  \ \
 \__|__/\_____/ \_____/_/  =  \_\_|  |_|
```
It's one line higher (and a few characters wider), but I think the S looks better with the extra line.
Okay, back to work for me... :)

**Comment by paemurru** [comment]
Together with the rest of the banner text, the design by @joschmitt does not fit in a 80-character width terminal, which is a big down-side

**Comment by joschmitt** [comment]
> Together with the rest of the banner text, the design by @joschmitt does not fit in a 80-character width terminal, which is a big down-side

Yeah, I think the "fine-print" can also come below the logo. But yes, if we want to go full julia-style this doesn't work.

**Comment by paemurru** [comment]
Removed top row dot character as recommended by @lgoettgens
banner_6_dot

**Comment by JohnAAbbott** [comment]
To my eyes the current (old) banner is the worst: all the newer proposals are better!
I understand the reason for wanting the vertical lines in the banner (to mimic those in the logo), but to my eyes they look like "something went wrong" in the ASCII art version.  TBH, it is not that important to me.
Where do we vote?

**Comment by lgoettgens** [comment]
> Where do we vote?

@paemurru will create a poll in this thread.

**Comment by HereAround** [comment]
> > Where do we vote?
> 
> @paemurru will create a poll in this thread.

Despite reading the corresponding message on slack, I am uncertain how the vote is supposed to work. So...

I like `Julia-style, clean version with no vertical bars` best.

Thank you for working on this!

(I agree with @JohnAAbbott. Also to my eyes, the vertical lines in the third banner look like "something went wrong" in the ASCII art version.)

**Comment by paemurru** [comment]
@HereAround Thank you! I would not have noticed myself that the vertical lines look odd.

How to vote: at the end of the first post of this pull request, there is a button to add a reaction.
scrn1 b
To vote, put a reaction according to which banner you wish to vote for.
scrn2 b

**Comment by wdecker** [comment]
After looking again at the two julia like designs, I believe that the O and/or the distance to the S are too small. Von meinem iPhone gesendetAm 24.02.2024 um 11:32 schrieb Erik Paemurru ***@***.***>:﻿
@HereAround Thank you! I would not have noticed myself that the vertical lines look odd.
How to vote: at the end of the first post of this pull request, there is a button to add a reaction.
scrn1.b.png (view on web)
To vote, put a reaction according to which banner you wish to vote for.
scrn2.b.png (view on web)

**Comment by paemurru** [comment]
@wdecker Do you think this is better?

```
  ___   ____   ____    _    ____  
 / _ \ / ___| / ___|  / \  |  _ \ 
| | | |\___ \| |     / _ \ | |_) |
| |_| | ___) | |___ / ___ \|  _ < 
 \___/ |____/ \____/_/   \_\_| \_\
```

**Comment by Krastanov** [comment]
Just for reference, a similar question and a lot of banner-handling code is being discussed for julia itself: https://github.com/JuliaLang/julia/pull/51811

In particular, the new StyledStrings standard library can be pretty helpful.

**Comment by paemurru** [comment]
Increased spacing between O and S according to suggestion by @wdecker.
```
  ___   ____   ____    _    ____
 / _ \ / ___| / ___|  / \  |  _ \   |  Combining ANTIC, GAP, Polymake, Singular
| | | |\___ \| |     / _ \ | |_) |  |  Type "?Oscar" for more information
| |_| | ___) | |___ / ___ \|  _ <   |  Manual: https://docs.oscar-system.org
 \___/ |____/ \____/_/   \_\_| \_\  |  Version 1.1.0-DEV
```

Added a one-line banner for small terminals (width less than 79 characters)
```
OSCAR 1.1.0-DEV  https://docs.oscar-system.org  Type "?Oscar" for help
```

**Pull request merged**